src/handlers/InstanceHandler.ts

```diff
@@ -4,99 +4,194 @@ import MetagameEventEvent from './census/events/MetagameEventEvent';
 import {jsonLogOutput} from '../utils/json';
 import {getLogger} from '../logger';
 import ApplicationException from '../exceptions/ApplicationException';
-import {InstanceSchemaInterface} from '../models/InstanceModel';
 import {MetagameEventState} from '../constants/metagameEventState';
-import {instanceId} from '../utils/instance';
 import MongooseModelFactory from '../factories/MongooseModelFactory';
 import {TYPES} from '../constants/types';
-import ActiveInstanceAuthority from '../authorities/ActiveInstanceAuthority';
+import {World} from '../constants/world';
+import PS2AlertsInstanceInterface from '../interfaces/PS2AlertsInstanceInterface';
+import PS2AlertsMetagameInstance from '../instances/PS2AlertsMetagameInstance';
+import {Zone} from '../constants/zone';
+import {InstanceMetagameSchemaInterface} from '../models/instance/InstanceMetagame';
+import {InstanceCustomWorldZoneSchemaInterface} from '../models/instance/InstanceCustomWorldZone';
+import {Ps2alertsEventState} from '../constants/ps2alertsEventState';
+import _ from 'lodash';
 
 @injectable()
 export default class InstanceHandler implements InstanceHandlerInterface {
 
     private static readonly logger = getLogger('InstanceHandler');
 
-    private readonly factory: MongooseModelFactory<InstanceSchemaInterface>;
+    private readonly currentInstances: PS2AlertsInstanceInterface[] = [];
 
-    private readonly activeInstanceAuthority: ActiveInstanceAuthority;
+    private readonly instanceMetagameModelFactory: MongooseModelFactory<InstanceMetagameSchemaInterface>;
+    private readonly instanceCustomWorldZoneInstanceModelFactory: MongooseModelFactory<InstanceCustomWorldZoneSchemaInterface>;
 
-    constructor(@inject(TYPES.instanceModelFactory) factory: MongooseModelFactory<InstanceSchemaInterface>,
-                                                    activeInstanceAuthority: ActiveInstanceAuthority,
+    constructor(
+    @inject(TYPES.instanceMetagameModelFactory) instanceMetagameModelFactory: MongooseModelFactory<InstanceMetagameSchemaInterface>,
+        @inject(TYPES.instanceCustomWorldZoneModelFactory) instanceCustomWorldZoneInstanceModelFactory: MongooseModelFactory<InstanceCustomWorldZoneSchemaInterface>,
     ) {
-        this.factory = factory;
-        this.activeInstanceAuthority = activeInstanceAuthority;
+        this.instanceMetagameModelFactory = instanceMetagameModelFactory;
+        this.instanceCustomWorldZoneInstanceModelFactory = instanceCustomWorldZoneInstanceModelFactory;
+        void this.init();
```

Comment:
Maybe add this to a boot method of a service?

Reply:
Yeah, thought about doing that